apis/database/v1alpha1/postgres.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PostgresParameters define the desired state of an AWS IAM Role.
+type PostgresParameters struct {
+
+	// DatabaseSize is the size of the database in a valid Go notation
+	// e.g., 1Gi
+	DatabaseSize string `json:"databaseSize"`
+
+	// MasterUsername is the name for the master user.
+	// Constraints:
+	//    * Required for PostgreSQL.
+	//    * Must be 1 to 63 letters or numbers.
+	//    * First character must be a letter.
+	//    * Cannot be a reserved word for the chosen database engine.
+	// +immutable
+	// +optional
+	MasterUsername *string `json:"masterUsername,omitempty"`
+
+	// Database specifies the default database to be created with the image
+	// +optional
+	Database *string `json:"database,omitempty"`
+
+	// StorageClass specifies the storage classed used for the PVC.
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+
+	// Port is the port number on which Postgres will listen for connections.
+	// +optional
+	Port *int `json:"port,omitempty"`
```

Comment:
Are these fields mutable?

Reply:
Nope! I'll add the immutable label to these fields